gridstatus/pjm.py

```diff
@@ -538,6 +598,446 @@ def _get_key(self):
 
         return settings["subscriptionKey"]
 
+    @staticmethod
+    def _new_dv_session(session, verbose=False):
+        if verbose:
+            print(f"GET {DATAVIEWER_LMP_URL}")
+        response = session.get(DATAVIEWER_LMP_URL)
+
+        html = response.content
+        doc = bs4.BeautifulSoup(html, "html.parser")
+
+        scripts = doc.find_all("script", {"nonce": True})
+
+        nonce = None
+        for script in scripts:
+            nonce = script["nonce"]
+            if nonce is not None:
+                break
+
+        view_state = None
+        view_states = doc.find_all(
+            "input",
+            {"type": "hidden", "name": "javax.faces.ViewState"},
+        )
+        for view_state in view_states:
+            view_state = view_state["value"]
+            if view_state is not None:
+                break
+
+        if nonce is None or view_state is None:
+            raise ValueError("Could not create new DV Session")
+        else:
+            return {
+                "nonce": nonce,
+                "view_state": view_state,
+                "session": session,
+            }
+
+    def _dv_lmp_fetch_data(self, session, verbose=False):
+        initial_fetch = session.get(DATAVIEWER_LMP_URL)
```

Comment:
I see you also make `session.get(DATAVIEWER_LMP_URL)` inside of `_new_dv_session`. do you need it in both places or should you be passing initial_fetch to that function instead?